crates/offchain-db/src/postgres_connector.rs

```diff
@@ -0,0 +1,217 @@
+use postgres::Error;
+use tokio_postgres::{Client, NoTls, Row};
+
+use crate::config::OffchainDbConfig;
+use crate::tables::{DbSequencerCommitment, Tables, SEQUENCER_COMMITMENT_TABLE_CREATE_QUERY};
+use crate::utils::get_db_extension;
+
+pub struct PostgresConnector {
+    client: Client,
+}
+
+impl PostgresConnector {
+    pub async fn new(pg_config: OffchainDbConfig) -> Result<Self, Error> {
+        let (client, connection) =
+            tokio_postgres::connect(pg_config.parse_to_connection_string().as_str(), NoTls).await?;
+
+        // The connection object performs the actual communication with the database,
+        // so spawn it off to run on its own.
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                eprintln!("connection error: {}", e);
+            }
+        });
+        // create new db
+        let db_name = format!("citrea{}", get_db_extension());
+        let _ = client
+            .batch_execute(&format!("CREATE DATABASE {};", db_name.clone()))
+            .await;
```

Comment:
create database if not exists ?

what happens when I close the sequencer and rerun it?

Reply:
> create database if not exists ?

There is not really a goodway to do that without workarounds, in this scenario if the db exists it returns error but it is ignored